async-nats/src/auth_utils.rs

```diff
@@ -0,0 +1,95 @@
+// Copyright 2020-2022 The NATS Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use nkeys::KeyPair;
+use once_cell::sync::Lazy;
+use regex::Regex;
+use std::{io, path::PathBuf};
+
+/// Loads user credentials file with jwt and key
+pub(crate) async fn load_creds(path: PathBuf) -> io::Result<(String, KeyPair)> {
+    tokio::task::spawn_blocking(move || {
+        let contents = std::fs::read_to_string(&path).map_err(|err| {
+            io::Error::new(
+                io::ErrorKind::Other,
+                format!("loading creds file '{}': {}", path.display(), err),
+            )
+        })?;
+        jwt_kp(&contents)
+    })
+    .await?
+}
+
+pub(crate) fn jwt_kp(contents: &str) -> io::Result<(String, KeyPair)> {
```

Comment:
I would avoid abbrevations like this and call it `jwt_key_pair`.

Reply:
done